plugins/store/src/store.rs

```diff
@@ -550,11 +547,9 @@ impl<R: Runtime> Store<R> {
     pub fn close_resource(&self) {
         let store = self.store.lock().unwrap();
         let app = store.app.clone();
-        let state = app.state::<StoreState>();
-        let stores = state.stores.lock().unwrap();
-        if let Some(rid) = stores.get(&store.path).copied() {
-            drop(store);
-            drop(stores);
+        let state = app.state::<StoreState<R>>();
+        let mut stores = state.stores.lock().unwrap();
+        if let Some(rid) = validate_resource_id(&app, stores.get_mut(&store.path)) {
             let _ = app.resources_table().close(rid);
```

Comment:
This is technically not safe since we didn't hold a lock through out and the resource id can be reused (https://github.com/tauri-apps/tauri/pull/9272) ...